### ISSUE TITLE
Compile error on Windows with MSYS2 (__cpuid(..) too few arguments)

### ISSUE THREAD

**Issue by cwansart** [ogdf/ogdf#23]
I was trying to compile ogdf on Windows with gcc from MSYS2 (http://www.msys2.org/). It caused an error in src/ogdf/basic/System.cpp on line 77:

    #if defined(OGDF_SYSTEM_WINDOWS)
        __cpuid(CPUInfo, infoType);
    #else

This version is from MSVC but the definition of OGDF_SYSTEM_WINDOWS doesn't cover that. So I added another condition:

    #if defined(OGDF_SYSTEM_WINDOWS) && defined(_MSC_VER)
        __cpuid(CPUInfo, infoType);
    #else

This version succesfully builds.

**Comment by sbeyer** [comment]
It seems msys2 expects the POSIX/GNU-alike headers and should be treated like `OGDF_SYSTEM_UNIX`. If this is the case for all msys installations, it should solve the problem to test for a certain macro to enable `OGDF_SYSTEM_UNIX` (it may be debatable if this is the right way to go but it seems that systems like msys "emulate" Unix in a certain way). Is one of the mingw macros `__MINGW32__` or `__MINGW64__` set on msys?

**Comment by cwansart** [comment]
No, both are not set. I checked it via `cc -m64 -dM -E -x c /dev/null | grep "MINGW"`

**Comment by sbeyer** [comment]
Hmm, it's probably the best way to add a test program to cmake calling `cpuid` to tell what way to go. But I hoped there'd be a clearer way.

**Comment by sbeyer** [comment]
Maybe it was a mistake to judge the problem from afar. Looking at the code, there is an inconsistency.

Does it help if you *only* change line 63 of System.cpp from
`#if defined(_MSC_VER)` to `#if defined(OGDF_SYSTEM_WINDOWS)`?

**Comment by sbeyer** [comment]
Even more probably, it is vice versa: changing line 76 to `#if defined(_MSC_VER)` could help because `intrin.h` seems to be MSVC-specific, not Windows-specific. Could you please test? Maybe you have to change other `#ifdef`/`#if defined()`, too.
